src/functional/iterator/ir.py

```diff
@@ -71,9 +71,14 @@ class FunctionDefinition(Node, SymbolTableTrait):
 class StencilClosure(Node):
     domain: Expr
     stencil: Expr
-    output: SymRef  # we could consider Expr for cases like make_tuple(out0,out1)
+    output: Union[SymRef, FunCall]
     inputs: List[SymRef]
 
+    @datamodels.validator("output")
```

Comment:
I am happy to reopen this discussion again, but validation should ever only happen in the passes. See here for my alternative proposal https://github.com/havogt/gt4py/pull/29 which coincidentally didn't land in the PR merged into here this morning.

Reply:
I don't see any argument for not adding this here. Can you give me one? I also don't see something in https://github.com/havogt/gt4py/pull/29 that is related.

Reply:
Not one I agree with beside that we strictly follow this convention. If I remember correctly @egparedes and/or @DropD main argument is that the validation code was complex in GTC and contained to much logic making the IRs hard to read. I agree with that observation, but not the conclusion. For me the IRs were to complex leading to complex validators, but that is not an argument for not having validators.

Reply:
I think this convention doesn't apply to field validators, but only to root validators, especially the ones that did passes. I have a strong opinion that this is the correct thing to do here.

Reply:
I have the same strong opinion so I am fine with merging, but please check with @egparedes after he is back.